src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void handle(List<FruitTransaction> transactions, Map<String, Integer> inventory) {
+        for (FruitTransaction transaction : transactions) {
+            if (transaction.getOperation() == FruitTransaction.Operation.PURCHASE) {
+                inventory.merge(transaction.getFruit(), -transaction.getQuantity(), Integer::sum);
```

Comment:
Ensure that the resulting inventory quantity does not become negative. You might want to add a check to prevent subtracting more than is available, which could lead to negative inventory values.